test/extended/authorization/tolerations_restriction.go

```diff
@@ -0,0 +1,136 @@
+package authorization
+
+import (
+	"fmt"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+var _ = g.Describe("[Feature: Toleration restrictions] Tolerations restrictions should be functional", func() {
+	defer g.GinkgoRecover()
+	f := e2e.NewDefaultFramework("tolerations-restriction")
+	cs, err := e2e.LoadClientset()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	g.Describe("Pod with service account", func() {
+		g.It(fmt.Sprintf("with enough privileges should pass"), func() {
+			err := createClusterRole(cs, f.Namespace.Name)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = createSAWithTolerationsAccess(cs, f.Namespace.Name)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = createRoleBinding(cs, f.Namespace.Name)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = createPodWithSA(cs, f.Namespace.Name)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+	})
+	g.Describe("Pod without service account should fail", func() {
```

Comment:
This test doesn't do what you want. All pods have serviceaccounts enforced via admission.  You need to test and SA without permissions.

Reply:
True but the plugin also enforces the existence of SA, so this case is still valid, I'll add one more test with a SA without enough privileges but I wanted to make sure that the happy path works fine.